src/client/app/effects/snippet.ts

```diff
@@ -372,15 +361,45 @@ export class SnippetEffects {
         properties['mode'] = isViewMode ? 'view' : 'editor';
         AI.trackEvent(mode, properties);
 
-        /*
-         * If a imported snippet is a SAMPLE or the app is in view mode, then skip the save (simply to avoid clutter).
-         * The snippet will get saved as soon as the user makes any changes (if in editor mode).
+        /**
+         * If the action here involves true importing rather than re-opening,
+         * and if the name is already taken by a local snippet, ask the user if they want to:
+         * import the snippet with a new name, go to the already existing snippet, or cancel the import
          */
-        if (mode !== Snippet.ImportType.SAMPLE && !isViewMode) {
-            actions.push(new Snippet.SaveAction(snippet));
+        let importResult = null;
+        if (mode !== Snippet.ImportType.OPEN && this._exists(snippet.name)) {
+            importResult = await this._uiEffects.alert(
+                Strings().snippetNameDuplicationError,
+                `${Strings().importButtonLabel} ${snippet.name}`,
+                Strings().snippetImportExistingButtonLabel, Strings().defaultSnippetTitle, Strings().cancelButtonLabel /* user options */
+            );
         }
 
-        return Observable.from(actions);
+        let actions: Action[] = [];
+        if (importResult === Strings().cancelButtonLabel) {
+            return Observable.from([]);
+        } else if (importResult === Strings().snippetImportExistingButtonLabel) {
+            storage.snippets.values().some(item => {
+                if (item.name.trim() === snippet.name.trim()) {
+                    actions.push(new Snippet.ImportSuccessAction(item));
+                }
+                return item.name.trim() === snippet.name.trim();
+            });
+            return Observable.from(actions);
```

Comment:
Why do you need this return in every block. After the if/else blocks why not just do a return Observable.from(actions);